saleor/checkout/utils.py

```diff
@@ -175,7 +175,11 @@ def get_or_create_user_cart(user, cart_queryset=Cart.objects.all()):
     defaults = {
         'shipping_address': user.default_shipping_address,
         'billing_address': user.default_billing_address}
-    return cart_queryset.get_or_create(user=user, defaults=defaults)[0]
+
+    cart = cart_queryset.filter(user=user).first()
```

Comment:
Wouldn't we achieve the same with passing `user` to `defaults` in the original implementation?

Reply:
```get_or_create``` can not be used since it will raise ```MultipleObjectsReturned``` error. We should use ```filter``` instead.

Reply:
I feel like that's a workaround, and we should just check all the places where one can create a cart, and make sure that there's only one present at all the time.

Reply:
I think there is nothing wrong if a user has multiple carts? We only need to make sure in storefront or API only the latest changed cart is returned when accessing user cart.

Reply:
I've looked into it and this is the right place to fix the issue, but the problem is that we're using this function only in Storefront 1.0 and in API we duplicate logic for creating cart instances. We have to try to reuse the core logic as much as possible. Based on your implementation @jxltom I'll try to refactor it a bit more since I'm already into it.

Reply:
Also I don't think that we shuold allow creating more than one cart instance for a user at once. Is there any use-case for that? It would be much simpler to assume that user can have only one and optimize the code aginst that case.

Reply:
No problem for that.

As for multiple cart for single user, I can't think of any user cases for now but it may require some work to make sure every user only has one cart since ```checkoutCreate``` or ```CheckoutCustomerAttach``` or storefront 1.0 can all create checkout for users. We need to adjust all of these APIs.

On the other way, I can't think of any harm if a user has multiple carts. We only need to make sure we always use the last changed cart. It makes the life easier.

Reply:
About your first remark - that's the current problem. There are multiple places in our code where a new cart instance is created but we're not reusing the same logic which may lead to bugs and code duplication.

Reply:
Yes, that requires some enhancement if we only allow user has one cart.